pkg/agent/openflow/pipeline.go

```diff
@@ -114,111 +114,92 @@ func (a ofAction) String() string {
 	}
+
+	// ofTableCache caches the OpenFlow tables used in the pipeline, and it supports using the table ID and name as the index to query the OpenFlow table.
+	ofTableCache = cache.NewIndexer(tableIDIndexFunc, cache.Indexers{tableNameIndex: tableNameIndexFunc})
 )
 
+func tableNameIndexFunc(obj interface{}) ([]string, error) {
+	ofTable := obj.(binding.Table)
+	return []string{ofTable.GetName()}, nil
+}
+
+func tableIDIndexFunc(obj interface{}) (string, error) {
+	ofTable := obj.(binding.Table)
+	return fmt.Sprintf("%d", ofTable.GetID()), nil
+}
+
+func getTableByID(id uint8) binding.Table {
+	obj, exists, _ := ofTableCache.GetByKey(fmt.Sprintf("%d", id))
+	if !exists {
+		return nil
+	}
+	return obj.(binding.Table)
+}
+
 // GetFlowTableName returns the flow table name given the table number. An empty
 // string is returned if the table cannot be found.
-func GetFlowTableName(tableNumber binding.TableIDType) string {
-	for _, t := range FlowTables {
-		if t.Number == tableNumber {
-			return t.Name
-		}
+func GetFlowTableName(tableNumber uint8) string {
+	table := getTableByID(tableNumber)
+	if table == nil {
+		return ""
 	}
-	return ""
+	return table.GetName()
 }
 
 // GetFlowTableNumber does a case insensitive lookup of the table name, and
 // returns the flow table number if the table is found. Otherwise TableIDAll is
 // returned if the table cannot be found.
-func GetFlowTableNumber(tableName string) binding.TableIDType {
-	for _, t := range FlowTables {
-		if strings.EqualFold(t.Name, tableName) {
-			return t.Number
-		}
+func GetFlowTableNumber(tableName string) uint8 {
+	objs, _ := ofTableCache.ByIndex(tableNameIndex, tableName)
+	if len(objs) == 0 {
+		return binding.TableIDAll
+	}
+	return objs[0].(binding.Table).GetID()
+}
+
+func PrintTableBaseInfo() string {
+	msg := ""
+	for _, obj := range ofTableCache.List() {
+		t := obj.(binding.Table)
+		msg += fmt.Sprintf("\n  %d\t%s", uint32(t.GetID()), t.GetName())
 	}
-	return binding.TableIDAll
+	return msg
 }
 
-func GetAntreaPolicyEgressTables() []binding.TableIDType {
-	return []binding.TableIDType{
+func GetAntreaPolicyEgressTables() []binding.Table {
+	return []binding.Table{
 		AntreaPolicyEgressRuleTable,
 		EgressDefaultTable,
 	}
 }
 
-func GetAntreaPolicyIngressTables() []binding.TableIDType {
-	return []binding.TableIDType{
+func GetAntreaPolicyIngressTables() []binding.Table {
+	return []binding.Table{
 		AntreaPolicyIngressRuleTable,
 		IngressDefaultTable,
 	}
 }
 
-func GetAntreaPolicyBaselineTierTables() []binding.TableIDType {
-	return []binding.TableIDType{
-		EgressDefaultTable,
-		IngressDefaultTable,
+func GetAntreaPolicyBaselineTierTables() []uint8 {
+	return []uint8{
+		EgressDefaultTable.GetID(),
+		IngressDefaultTable.GetID(),
 	}
 }
 
-func GetAntreaPolicyMultiTierTables() []binding.TableIDType {
-	return []binding.TableIDType{
-		AntreaPolicyEgressRuleTable,
-		AntreaPolicyIngressRuleTable,
+func GetAntreaPolicyMultiTierTables() []uint8 {
```

Comment:
`GetAntreaPolicyEgressTables` and `GetAntreaPolicyIngressTables` return tables while `GetAntreaPolicyBaselineTierTables` and `GetAntreaPolicyMultiTierTables` return table IDs. Can we unify the return value for similar functions, or name the latter ones `GetXXXTableIDs`?

Reply:
Updated.